specification/ml/put_trained_model/MlPutTrainedModelRequest.ts

```diff
@@ -17,21 +17,36 @@
  * under the License.
  */
 
+import { UserDefinedValue } from '@spec_utils/UserDefinedValue'
+import { InferenceConfigContainer } from '@_types/aggregations/pipeline'
 import { RequestBase } from '@_types/Base'
+import { Id } from '@_types/common'
+import { Definition, Input } from './types'
 
 /**
  * @rest_spec_name ml.put_trained_model
  * @since 7.10.0
- * @stability TODO
+ * @stability stable
  */
 export interface Request extends RequestBase {
   path_parts: {
-    stub: string
-  }
-  query_parameters?: {
-    stub?: string
+    /** The unique identifier of the trained model. */
+    model_id: Id
   }
   body?: {
-    stub?: string
+    /** The compressed (GZipped and Base64 encoded) inference definition of the model. If compressed_definition is specified, then definition cannot be specified. */
+    compressed_definition?: string
+    /** The inference definition for the model. If definition is specified, then compressed_definition cannot be specified. */
+    definition?: Definition
```

Comment:
Can we introduce a union type to handle these mutually exclusive properties?

Reply:
Since it's already explained in the property descriptions for now we prefer to keep it this way. We still need to figure out an elegant way to address this case that works well in every downstream client.